### PR TITLE
Preview Release ConnectedMachine 2024-07-31

### PR DESCRIPTION
**Pull request by yaotongms**
## Description

General release on ConnectedMachine that includes the following new cmdlets:
- New-AzConnectedLicenseProfile: create a new license profile for enrolling arc machine to PayGo or Self Attestation
- Update-AzConnectedLicenseProfile: update the license profile
- Get-AzConnectedLicenseProfile: get info on the license profile
- Remove-AzConnectedLicenseProfile: remove the license profile

## Mandatory Checklist

- Please choose the target release of Azure PowerShell. (⚠️**Target release** is a different concept from **API readiness**. Please click below links for details.)
  - [x] [General release](../blob/main/CONTRIBUTING.md#target-release-types)
  - [ ] [Public preview](../blob/main/CONTRIBUTING.md#target-release-types)
  - [ ] [Private preview](../blob/main/CONTRIBUTING.md#target-release-types)
  - [ ] [Engineering build](../blob/main/CONTRIBUTING.md#target-release-types)
  - [ ] No need for a release

- [x] Check this box to confirm: **I have read the [_Submitting Changes_](../blob/main/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/main/CONTRIBUTING.md) and reviewed the following information:**

* **SHOULD** update `ChangeLog.md` file(s) appropriately
    * For SDK-based development mode, update `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`.
        * A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header in the past tense. 
    * For autorest-based development mode, include the changelog in the PR description.
    * Should **not** change `ChangeLog.md` if no new release is required, such as fixing test case only.
* **SHOULD** regenerate markdown help files if there is cmdlet API change. [Instruction](../blob/main/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)
* **SHOULD** have proper test coverage for changes in pull request.
* **SHOULD NOT** adjust version of module manually in pull request

### CODE REVIEW

**Comment by vidai-msft** [comment]
@yaotongms Please fill in the examples and test cases for the cmdlets.

**Comment by vidai-msft** [comment]
@yaotongms Please kindly note there are breaking changes detected in the static analysis. Please confirm if these will really cause breaking on custom scripts. If so, may I know if you have submitted pre-announcements?

**Comment by yaotongms** [comment]
> @yaotongms Please fill in the examples and test cases for the cmdlets.

examples and tests updated

**Comment by yaotongms** [comment]
> @yaotongms Please kindly note there are breaking changes detected in the static analysis. Please confirm if these will really cause breaking on custom scripts. If so, may I know if you have submitted pre-announcements?

they shouldn't cause any breaking on custom scripts.

**Comment by yaotongms** [comment]
changelog: Updated preview version api of HybridCompute to 2024-07-31

**Comment by yaotongms** [comment]
swagger: https://github.com/Azure/azure-rest-api-specs/tree/main/specification/hybridcompute/resource-manager/Microsoft.HybridCompute/preview/2024-07-31-preview

**Comment by vidai-msft** [comment]
@yaotongms Please check the issues detected in CI. Please kindly note the code freeze date is 11/11 6PM PST.

File: src/ConnectedMachine/ConnectedMachine.Autorest/test/Get-AzConnectedNetworkSecurityPerimeterConfiguration.Tests.ps1 [hunk 1]
```diff
@@ -16,12 +16,12 @@ if(($null -eq $TestName) -or ($TestName -contains 'Get-AzConnectedNetworkSecurit
...
 Describe 'Get-AzConnectedNetworkSecurityPerimeterConfiguration' {
     It 'List' {
-        $all = @(Get-AzConnectedNetworkSecurityPerimeterConfiguration -ResourceGroupName $env.ResourceGroupNameNSP -ScopeName $env.PrivateLinkScopeNameNSP)
+        $all = @(Get-AzConnectedNetworkSecurityPerimeterConfiguration -ResourceGroupName $env.ResourceGroupName -ScopeName $env.PrivateLinkScopeName)
         $all | Should -Not -BeNullOrEmpty
     }
 
-    It 'Get' {
-        $all = @(Get-AzConnectedNetworkSecurityPerimeterConfiguration -ResourceGroupName $env.ResourceGroupNameNSP -ScopeName $env.PrivateLinkScopeNameNSP -PerimeterName $env.PerimeterName)
+    It 'Get' -skip {
```
Comment (vidai-msft) [inline src/ConnectedMachine/ConnectedMachine.Autorest/test/Get-AzConnectedNetworkSecurityPerimeterConfiguration.Tests.ps1:23]: Why this test case is marked as skip?

File: src/ConnectedMachine/ConnectedMachine.Autorest/test/New-AzConnectedLicenseProfile.Tests.ps1 [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+        # SA benefit only
+        # $all = @(New-AzConnectedLicenseProfile -MachineName $env.MachineNameSA -ResourceGroupName $env.ResourceGroupNameProfile -Location $env.Location -SoftwareAssuranceCustomer)
+
+        # WS paygo and hotpatch
+        $all = @(New-AzConnectedLicenseProfile -MachineName $env.MachineNamePaygo -ResourceGroupName $env.ResourceGroupNameProfile -Location $env.Location -ProductProfileProductType "WindowsServer" -ProductProfileSubscriptionStatus "Enabled" -ProductProfileProductFeature $productfeature)
+        $all | Should -Not -BeNullOrEmpty
+    }
+
+    It 'CreateViaJsonString' -skip {
```
Comment (vidai-msft) [inline src/ConnectedMachine/ConnectedMachine.Autorest/test/New-AzConnectedLicenseProfile.Tests.ps1:30]: If other parameter sets are not necessary, please remove the `It` sections.

File: src/ConnectedMachine/ConnectedMachine.Autorest/test/New-AzConnectedLicenseProfileFeature.Tests.ps1 [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+  while(-not $mockingPath) {
+      $mockingPath = Get-ChildItem -Path $currentPath -Recurse -Include 'HttpPipelineMocking.ps1' -File
+      $currentPath = Split-Path -Path $currentPath -Parent
+  }
+  . ($mockingPath | Select-Object -First 1).FullName
+}
+
+Describe 'New-AzConnectedLicenseProfileFeature' {
+    It '__AllParameterSets' -skip {
+        { throw [System.NotImplementedException] } | Should -Not -Throw
```
Comment (vidai-msft) [inline src/ConnectedMachine/ConnectedMachine.Autorest/test/New-AzConnectedLicenseProfileFeature.Tests.ps1:19]: Please add a test case for this cmdlet.

File: src/ConnectedMachine/ConnectedMachine.Autorest/test/New-AzConnectedLicenseProfileUpdate.Tests.ps1 [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+  while(-not $mockingPath) {
+      $mockingPath = Get-ChildItem -Path $currentPath -Recurse -Include 'HttpPipelineMocking.ps1' -File
+      $currentPath = Split-Path -Path $currentPath -Parent
+  }
+  . ($mockingPath | Select-Object -First 1).FullName
+}
+
+Describe 'New-AzConnectedLicenseProfileUpdate' {
+    It '__AllParameterSets' -skip {
+        { throw [System.NotImplementedException] } | Should -Not -Throw
```
Comment (vidai-msft) [inline src/ConnectedMachine/ConnectedMachine.Autorest/test/New-AzConnectedLicenseProfileUpdate.Tests.ps1:19]: Please add a test case for this cmdlet.

File: src/ConnectedMachine/ConnectedMachine.Autorest/test/New-AzConnectedMachineExtension.Tests.ps1 [hunk 5]
```diff
@@ -13,7 +13,7 @@ while(-not $mockingPath) {
 
 Import-Module "$PSScriptRoot/helper.psm1" -Force
 
-Describe 'New-AzConnectedMachineExtension' {
+Describe 'New-AzConnectedMachineExtension' -{
```
Comment (vidai-msft) [inline src/ConnectedMachine/ConnectedMachine.Autorest/test/New-AzConnectedMachineExtension.Tests.ps1:16]: Is this dash by mistake?

File: src/ConnectedMachine/ConnectedMachine.Autorest/test/New-AzConnectedMachineRunCommand.Tests.ps1 [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+  while(-not $mockingPath) {
+      $mockingPath = Get-ChildItem -Path $currentPath -Recurse -Include 'HttpPipelineMocking.ps1' -File
+      $currentPath = Split-Path -Path $currentPath -Parent
+  }
+  . ($mockingPath | Select-Object -First 1).FullName
+}
+
+Describe 'New-AzConnectedMachineRunCommand' {
+    It 'CreateExpanded' -skip {
+        { throw [System.NotImplementedException] } | Should -Not -Throw
```
Comment (vidai-msft) [inline src/ConnectedMachine/ConnectedMachine.Autorest/test/New-AzConnectedMachineRunCommand.Tests.ps1:19]: Please remove the place holders for the parameter sets that are not necessary.

File: src/ConnectedMachine/ConnectedMachine.Autorest/test/New-AzConnectedMachineRunCommand.Tests.ps1 [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+    }
+
+    It 'Create' {
+        $runCommand = New-AzConnectedMachineRunCommand -ResourceGroupName $env.ResourceGroupName -Location $env.Location -SourceScript $env.Script -RunCommandName $env.RunCommandName -MachineName $env.MachineName -Subscription $env.SubscriptionId
+        $runCommand.Count | Should -Not -BeNullOrEmpty
+    }
+
+    It 'CreateViaIdentityMachineExpanded' -skip {
+        { throw [System.NotImplementedException] } | Should -Not -Throw
```
Comment (vidai-msft) [inline src/ConnectedMachine/ConnectedMachine.Autorest/test/New-AzConnectedMachineRunCommand.Tests.ps1:36]: Please remove the place holders for the parameter sets that are not necessary.

File: src/ConnectedMachine/ConnectedMachine.Autorest/test/Remove-AzConnectedLicenseProfile.Tests.ps1 [hunk 8]
```diff
@@ -0,0 +1,25 @@
...
+  . ($mockingPath | Select-Object -First 1).FullName
+}
+
+Describe 'Remove-AzConnectedLicenseProfile' {
+    It 'Delete' {
+        Remove-AzConnectedLicenseProfile -MachineName $env.MachineNamePaygo -ResourceGroupName $env.ResourceGroupNameProfile 
+    }
+
+    It 'DeleteViaIdentity' -skip {
+        { throw [System.NotImplementedException] } | Should -Not -Throw
```
Comment (vidai-msft) [inline src/ConnectedMachine/ConnectedMachine.Autorest/test/Remove-AzConnectedLicenseProfile.Tests.ps1:23]: Please remove the place holders for the parameter sets that are not necessary.

File: src/ConnectedMachine/ConnectedMachine.Autorest/test/Remove-AzConnectedMachineRunCommand.Tests.ps1 [hunk 9]
```diff
@@ -0,0 +1,29 @@
...
+  . ($mockingPath | Select-Object -First 1).FullName
+}
+
+Describe 'Remove-AzConnectedMachineRunCommand' {
+    It 'Delete' {
+        Remove-AzConnectedMachineRunCommand -ResourceGroupName $env.ResourceGroupName -RunCommandName $env.RunCommandName -MachineName $env.MachineName
+    }
+
+    It 'DeleteViaIdentityMachine' -skip {
+        { throw [System.NotImplementedException] } | Should -Not -Throw
```
Comment (vidai-msft) [inline src/ConnectedMachine/ConnectedMachine.Autorest/test/Remove-AzConnectedMachineRunCommand.Tests.ps1:23]: Please remove the place holders for the parameter sets that are not necessary.

File: src/ConnectedMachine/ConnectedMachine.Autorest/test/Update-AzConnectedLicenseProfile.Tests.ps1 [hunk 10]
```diff
@@ -0,0 +1,43 @@
...
+
+Describe 'Update-AzConnectedLicenseProfile' {
+    It 'UpdateExpanded' {
+        $productfeature = New-AzConnectedLicenseProfileUpdate -Name "Hotpatch" -SubscriptionStatus "Enable"
+        $all = @(Update-AzConnectedLicenseProfile -MachineName $env.MachineNamePaygo -ResourceGroupName $env.ResourceGroupNameProfile -ProductProfileProductType "WindowsServer" -ProductProfileSubscriptionStatus "Enabled" -ProductProfileProductFeature $productfeature)
+        $all | Should -Not -BeNullOrEmpty
+    }
+
+    It 'UpdateViaJsonString' -skip {
+        { throw [System.NotImplementedException] } | Should -Not -Throw
```
Comment (vidai-msft) [inline src/ConnectedMachine/ConnectedMachine.Autorest/test/Update-AzConnectedLicenseProfile.Tests.ps1:25]: Please remove the place holders for the parameter sets that are not necessary.

File: src/ConnectedMachine/ConnectedMachine.Autorest/test/Update-AzConnectedMachineRunCommand.Tests.ps1 [hunk 11]
```diff
@@ -0,0 +1,35 @@
...
+  while(-not $mockingPath) {
+      $mockingPath = Get-ChildItem -Path $currentPath -Recurse -Include 'HttpPipelineMocking.ps1' -File
+      $currentPath = Split-Path -Path $currentPath -Parent
+  }
+  . ($mockingPath | Select-Object -First 1).FullName
+}
+
+Describe 'Update-AzConnectedMachineRunCommand' {
+    It 'UpdateExpanded' -skip {
+        { throw [System.NotImplementedException] } | Should -Not -Throw
```
Comment (vidai-msft) [inline src/ConnectedMachine/ConnectedMachine.Autorest/test/Update-AzConnectedMachineRunCommand.Tests.ps1:19]: Please remove the place holders for the parameter sets that are not necessary.

**Review by vidai-msft — CHANGES_REQUESTED** [review]
Please leave a message in this PR as the change log.

File: src/ConnectedMachine/ConnectedMachine.Autorest/test/Get-AzConnectedNetworkSecurityPerimeterConfiguration.Tests.ps1 (hunk 1 above)
Comment (yaotongms) [reply src/ConnectedMachine/ConnectedMachine.Autorest/test/Get-AzConnectedNetworkSecurityPerimeterConfiguration.Tests.ps1]: fixed

File: src/ConnectedMachine/ConnectedMachine.Autorest/test/New-AzConnectedLicenseProfile.Tests.ps1 (hunk 2 above)
Comment (yaotongms) [reply src/ConnectedMachine/ConnectedMachine.Autorest/test/New-AzConnectedLicenseProfile.Tests.ps1]: fixed

File: src/ConnectedMachine/ConnectedMachine.Autorest/test/New-AzConnectedLicenseProfileFeature.Tests.ps1 (hunk 3 above)
Comment (yaotongms) [reply src/ConnectedMachine/ConnectedMachine.Autorest/test/New-AzConnectedLicenseProfileFeature.Tests.ps1]: fixed

File: src/ConnectedMachine/ConnectedMachine.Autorest/test/New-AzConnectedLicenseProfileUpdate.Tests.ps1 (hunk 4 above)
Comment (yaotongms) [reply src/ConnectedMachine/ConnectedMachine.Autorest/test/New-AzConnectedLicenseProfileUpdate.Tests.ps1]: fixed

File: src/ConnectedMachine/ConnectedMachine.Autorest/test/New-AzConnectedMachineExtension.Tests.ps1 (hunk 5 above)
Comment (yaotongms) [reply src/ConnectedMachine/ConnectedMachine.Autorest/test/New-AzConnectedMachineExtension.Tests.ps1]: yes, fixed

File: src/ConnectedMachine/ConnectedMachine.Autorest/test/New-AzConnectedMachineRunCommand.Tests.ps1 (hunk 6 above)
Comment (yaotongms) [reply src/ConnectedMachine/ConnectedMachine.Autorest/test/New-AzConnectedMachineRunCommand.Tests.ps1]: fixed

File: src/ConnectedMachine/ConnectedMachine.Autorest/test/New-AzConnectedMachineRunCommand.Tests.ps1 (hunk 7 above)
Comment (yaotongms) [reply src/ConnectedMachine/ConnectedMachine.Autorest/test/New-AzConnectedMachineRunCommand.Tests.ps1]: fixed

File: src/ConnectedMachine/ConnectedMachine.Autorest/test/Remove-AzConnectedLicenseProfile.Tests.ps1 (hunk 8 above)
Comment (yaotongms) [reply src/ConnectedMachine/ConnectedMachine.Autorest/test/Remove-AzConnectedLicenseProfile.Tests.ps1]: fixed

File: src/ConnectedMachine/ConnectedMachine.Autorest/test/Remove-AzConnectedMachineRunCommand.Tests.ps1 (hunk 9 above)
Comment (yaotongms) [reply src/ConnectedMachine/ConnectedMachine.Autorest/test/Remove-AzConnectedMachineRunCommand.Tests.ps1]: fixed

File: src/ConnectedMachine/ConnectedMachine.Autorest/test/Update-AzConnectedLicenseProfile.Tests.ps1 (hunk 10 above)
Comment (yaotongms) [reply src/ConnectedMachine/ConnectedMachine.Autorest/test/Update-AzConnectedLicenseProfile.Tests.ps1]: fixed

File: src/ConnectedMachine/ConnectedMachine.Autorest/test/Update-AzConnectedMachineRunCommand.Tests.ps1 (hunk 11 above)
Comment (yaotongms) [reply src/ConnectedMachine/ConnectedMachine.Autorest/test/Update-AzConnectedMachineRunCommand.Tests.ps1]: fixed

File: src/ConnectedMachine/ConnectedMachine.Autorest/test/New-AzConnectedLicenseProfileFeature.Tests.ps1 [hunk 12]
```diff
@@ -0,0 +1,22 @@
...
+  $currentPath = $PSScriptRoot
+  while(-not $mockingPath) {
+      $mockingPath = Get-ChildItem -Path $currentPath -Recurse -Include 'HttpPipelineMocking.ps1' -File
+      $currentPath = Split-Path -Path $currentPath -Parent
+  }
+  . ($mockingPath | Select-Object -First 1).FullName
+}
+
+Describe 'New-AzConnectedLicenseProfileFeature' {
+    It '__AllParameterSets' -skip {
```
Comment (vidai-msft) [inline src/ConnectedMachine/ConnectedMachine.Autorest/test/New-AzConnectedLicenseProfileFeature.Tests.ps1:18]: Is this test case valid? If so, please remove the -skip.

Comment (yaotongms) [reply src/ConnectedMachine/ConnectedMachine.Autorest/test/New-AzConnectedLicenseProfileFeature.Tests.ps1]: fixed

**Review by vidai-msft — APPROVED** [review]
LGTM

**Pull request merged**